### PR TITLE
Setup user context

### PR DESCRIPTION
**Pull request by JoshL579**
`UserContextProvider` will fetch `userId` and `setUserId` in `state` everytime when user refresh the page. later private route only needs to check if userId is empty or not.

(This is task #38, sorry for the typo)

### CODE REVIEW

File: client/src/globals/UserContext.js [hunk 1]
```diff
@@ -8,6 +8,22 @@ export const UserContext = createContext({
...
+    useEffect(() => {
+        let status;
+        fetch("/getcookie", {
+            method: "GET",
+            headers: {
+                "Content-Type": "application/json"
+            }
+        })
+            .then(res => {
+                if (status === 200 && res.success === 'true') return setUserId(res.userId);
```
Comment (sina-jamshidi) [inline client/src/globals/UserContext.js]: shouldn't this be `res.status`?

File: server/api/jwt_check_handler.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+from utils.auth.token_generator import token_decoder
+
+token_check_handler = Blueprint('token_check_handler', __name__)
+
+
+@token_check_handler.route('/getcookie')
+def getcookie():
+    token = request.cookies.get('token')
+    if token is None:
+        return jsonify({'success': 'false', 'msg': 'Please log in'})
```
Comment (sina-jamshidi) [inline server/api/jwt_check_handler.py]: we can just return boolean. jsonify will handle converting it

**Review by sina-jamshidi — APPROVED** [review]

**Comment by fangyu3** [comment]
Hey Josh, i have some questions regarding this. Can you hold off on merging this? ill go over what we have on the backend and send the questions soon.

UPDATE:
Nevermind, I saw your private route branch and it answered the questions I had in mind.

**Review by fangyu3 — APPROVED** [review]

**Comment by JoshL579** [comment]
updated in PrivateRoute branch

**Pull request closed without merging**